From: David Daney <ddaney@caviumnetworks.com>
Date: Fri, 31 Oct 2008 18:23:09 +0000 (-0700)
Subject: MIPS: Return ENOSYS from sys32_syscall on 64bit kernels like elsewhere.
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=4fcface63ed4d71af1419d026821ed2f3983f3c1

MIPS: Return ENOSYS from sys32_syscall on 64bit kernels like elsewhere.

When the o32 errno was changed to ENOSYS, we forgot to update the code
for 64bit kernels.

Signed-off-by: David Daney <ddaney@caviumnetworks.com>
Acked-by: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit f18d0b1993f31e28eaf0e3b7e415788052985f95)
---

diff --git a/arch/mips/kernel/scall64-o32.S b/arch/mips/kernel/scall64-o32.S
index e10d5c0..b56d88f 100644
--- a/arch/mips/kernel/scall64-o32.S
+++ b/arch/mips/kernel/scall64-o32.S
@@ -196,7 +196,7 @@ LEAF(sys32_syscall)
 	jr	t2
 	/* Unreached */
 
-einval:	li	v0, -EINVAL
+einval:	li	v0, -ENOSYS
 	jr	ra
 	END(sys32_syscall)
 
